src/main/java/duke/Task.java

```diff
@@ -0,0 +1,76 @@
+package duke;
+
+import duke.parser.DukeParseException;
+import duke.parser.Parser;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public abstract class Task {
+    private final String desc;
+    private boolean isMarked;
+
+    public Task(String desc) {
+        this.desc = desc;
+        isMarked = false;
+    }
+
+    public static Task decode(String encodedTask) {
+        Pattern pattern = Pattern.compile("^(?<type>.)\\|(?<mark>.)\\|(?<taskString>.+)$");
+        Matcher matcher = pattern.matcher(encodedTask);
+        String type = matcher.group("type");
+        String mark = matcher.group("mark");
+        String input = matcher.group("taskString");
+        Task task;
+        switch (type) {
```

Comment:
Based on the SE-EDU Coding Standard, a Basic Coding Standard rule the `case`s in a switch-case statement should be on the same indentation level as the `switch` statement